content/en/blog/_posts/XXXX-XX-XX-fine-grained-supplementalgroups-control-ga.md

```diff
@@ -0,0 +1,168 @@
+
+## Fine-grained supplemental groups control in a Pod: `supplementaryGroupsPolicy`
+
+To tackle the above problem, Pod's `.spec.securityContext` now includes `supplementalGroupsPolicy` field.
+
+This field lets you control how Kubernetes calculates the supplementary groups for container processes within a Pod. The available policies are:
+
+* _Merge_: The group membership defined in `/etc/group` for the container's primary user will be merged. If not specified, this policy will be applied (i.e. as-is behavior for backward compatibility).
+
+* _Strict_: Only the group IDs specified in `fsGroup`, `supplementalGroups`, or `runAsGroup` are attached as supplementary groups to the container processes. Group memberships defined in `/etc/group` for the container's primary user are ignored.
+
+Let's see how `Strict` policy works. Below Pod manifest specifies `supplementalGroupsPolicy: Strict`:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: strict-supplementalgroups-policy
+spec:
+  securityContext:
+    runAsUser: 1000
+    runAsGroup: 3000
+    supplementalGroups: [4000]
+    supplementalGroupsPolicy: Strict
+  containers:
+  - name: ctr
+    image: registry.k8s.io/e2e-test-images/agnhost:2.45
+    command: [ "sh", "-c", "sleep 1h" ]
+    securityContext:
+      allowPrivilegeEscalation: false
+```
+
+The result of `id` command in the `ctr` container should be similar to this:
+
+```none
+uid=1000 gid=3000 groups=3000,4000
+```
+
+You can see `Strict` policy can exclude group `50000` from `groups`! 
+
+Thus, ensuring `supplementalGroupsPolicy: Strict` (enforced by some policy mechanism) helps prevent the implicit supplementary groups in a Pod.
+
+{{<note>}}
+A container with sufficient privileges can change its process identity. The `supplementalGroupsPolicy` only affect the initial process identity. See the following section for details.
+{{</note>}}
+
+## Attached process identity in Pod status
+
+This feature also exposes the process identity attached to the first container process of the container
+via `.status.containerStatuses[].user.linux` field. It would be helpful to see if implicit group IDs are attached.
+
+```yaml
+...
+status:
+  containerStatuses:
+  - name: ctr
+    user:
+      linux:
+        gid: 3000
+        supplementalGroups:
+        - 3000
+        - 4000
+        uid: 1000
+...
+```
+
+{{<note>}}
+Please note that the values in `status.containerStatuses[].user.linux` field is _the firstly attached_
+process identity to the first container process in the container. If the container has sufficient privilege
+to call system calls related to process identity (e.g. [`setuid(2)`](https://man7.org/linux/man-pages/man2/setuid.2.html), [`setgid(2)`](https://man7.org/linux/man-pages/man2/setgid.2.html) or [`setgroups(2)`](https://man7.org/linux/man-pages/man2/setgroups.2.html), etc.), the container process can change its identity. Thus, the _actual_ process identity will be dynamic.
+{{</note>}}
```

Comment:
can NRI plugin also change it or container runtime decide to fix it up somehow?

Reply:
Technically yes. NRI Plugin(via [`Hooks` in `ContainerAdjustment`](https://github.com/containerd/nri/blob/main/pkg/api/api.proto#L503)) or [RuntimeClass](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3619-supplemental-groups-policy#introducing-runtimeclass) can modify the underlying [`process.user.additionalGids`](https://github.com/opencontainers/runtime-spec/blob/main/config.md#user) in OCI Spec.